### DISCUSSION TITLE
Roots, of course ;)

### DISCUSSION THREAD

**Discussion by NagsFoV** [DynamicTreesTeam/DynamicTrees#967] [Ideas]
I'm surprised you went that far with all those superbs features without adding the roots. Those are half of a tree!
I know we take it for granted in video games, but trees have roots and because it's a voxel game, I say trees should have roots even in vanilla, even simple. It would be such a great feature for DynamicTree, roots growing in dirt as branches in air.
I'm sure you'll come with great result if you develop this, but here are some thoughts of mine:

-Roots grow in blocks like ores, keeping the initial block texture, addind roots in it (it even can be reversed if the tree dies)
-Roots can grow in stone, and other specified blocks but grow faster in dirt.
-Tree growing depends of roots size, so a tree in a single block of dirt won't grow up, and will remain small or grow slower in stone.
mining root would result to the tree's decay or falling.
-If roots are a separate thing as wood, Add-on mods could use roots for new usages.

Thank you for reading me

**Comment by supermassimo** [comment]
Adding roots to all trees would go beyond the scope of this mod, unfortunately. We have only implemented roots for trees that already have them (Mangrove trees behave exactly like you have suggested).
However you can use treepacks to add this behavour to other trees if youd like, as the logic is already present in the mod
